### PR TITLE
RFC - upgrade code location

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
RFC for discussion at tech sync for upgrade code location.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/80746a7f3462c2fa4ff15a92d6e196c67d652f6a'>80746a7f3462c2fa4ff15a92d6e196c67d652f6a</a></sub>
<div id="bundleSizeComment9423"/>

File: rfcs/react-components/convergence/upgrade-code-location.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+They have not yet asked for a package of shims to take a dependency on, but prefer to write their own shims if they are not upgrading to v9 directly.
+They have tended to author their own code-mod scripts.
+
+## Problem statement
+
+Where should upgrade helper code live?
+
+## Detailed Design or Proposal
+
+### Option A: Create upgrade packages in the fluent repo
```
Comment (micahgodbolt) [inline rfcs/react-components/convergence/upgrade-code-location.md:31]: Option A seems sound to keep complex dependencies out of react-components. Then we can build out a unique storybook for the upgrades and then use storybook composition to add these stories into our main storybook site

https://www.chromatic.com/docs/composition

Comment (jurokapsiar) [reply rfcs/react-components/convergence/upgrade-code-location.md:31]: this option is my personal favorite. keeps the structure clean and the storybook composition is something we have been planning to use for partners anyway.

**Review by jurokapsiar — APPROVED** [review]

File: rfcs/react-components/convergence/upgrade-code-location.md (hunk 1 above)
Comment (GeoffCoxMSFT) [reply rfcs/react-components/convergence/upgrade-code-location.md:31]: I agree that this is likely the best approach for this type of code.

**Review by micahgodbolt — APPROVED** [review]

**Comment by micahgodbolt** [comment]
wouldn't mind clarifying expected naming conventions. the 1st app has been merged, but no huge issue changing the name of the folder/package

**Pull request merged**